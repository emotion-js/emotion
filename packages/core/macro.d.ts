declare module '@emotion/core/macro' {
  export * from '@emotion/core'
}
