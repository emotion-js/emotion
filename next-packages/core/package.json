{
  "name": "@emotion/core",
  "version": "0.12.0",
  "main": "dist/core.cjs.js",
  "module": "dist/core.esm.js",
  "browser": {
    "./dist/core.cjs.js": "./dist/core.browser.cjs.js",
    "./dist/core.esm.js": "./dist/core.browser.esm.js"
  },
  "files": [
    "src",
    "dist"
  ],
  "author": "mitchellhamilton <mitchell@mitchellhamilton.me>",
  "license": "MIT",
  "dependencies": {
    "@emotion/cache": "^0.8.6",
    "@emotion/css": "^0.9.6",
    "@emotion/serialize": "^0.8.6",
    "@emotion/sheet": "^0.7.2",
    "@emotion/utils": "^0.8.0"
  },
  "peerDependencies": {
    "react": ">=16.3.0"
  },
  "devDependencies": {
    "html-tag-names": "^1.1.2",
    "svg-tag-names": "^1.1.1"
  },
  "repository": "https://github.com/emotion-js/emotion/tree/master/next-packages/core",
  "publishConfig": {
    "access": "public"
  },
  "umd:main": "./dist/index.min.js"
}
