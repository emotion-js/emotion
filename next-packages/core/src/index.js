// @flow
export { withCSSContext, Provider } from './context'
export { jsx } from './jsx'
export { Global } from './global'
export { keyframes } from './keyframes'
export { default as css } from '@emotion/css'
