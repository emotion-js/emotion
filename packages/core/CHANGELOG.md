# @emotion/core

## 10.0.21

### Patch Changes

- [7855db4a](https://github.com/emotion-js/emotion/commit/7855db4ae379f212e7b972b9108419154e17ed45) [#1537](https://github.com/emotion-js/emotion/pull/1537) Thanks [@Andarist](https://github.com/Andarist)! - Add missing `/* #__PURE__ */` annotation to creation of EmotionCssPropInternal

## 10.0.20

### Patch Changes

- [38bb2b19](https://github.com/emotion-js/emotion/commit/38bb2b19d4ff1552116829e068664516d167a3f8) [#1530](https://github.com/emotion-js/emotion/pull/1530) Thanks [@Andarist](https://github.com/Andarist)! - Fix class not being applied when putting nil as css prop value on wrapper component

## 10.0.17

### Patch Changes

- [66cda641](https://github.com/emotion-js/emotion/commit/66cda64128631790b81e3c9df273a972358ea593) [#1478](https://github.com/emotion-js/emotion/pull/1478) Thanks [@Andarist](https://github.com/Andarist)! - Add warnings about using illegal escape sequences
  - [10514a86](https://github.com/emotion-js/emotion/commit/10514a8635dcaa55b85c7bff90e2a9e14d1ba61f) [#1482](https://github.com/emotion-js/emotion/pull/1482) Thanks [@mitchellhamilton](https://github.com/mitchellhamilton)! - Remove react native bundles in favour of different isBrowser detection
  - [66cda641](https://github.com/emotion-js/emotion/commit/66cda64128631790b81e3c9df273a972358ea593) [#1478](https://github.com/emotion-js/emotion/pull/1478) Thanks [@Andarist](https://github.com/Andarist)! - Update Babel dependencies

## 10.0.16

### Patch Changes

- [47262b64](https://github.com/emotion-js/emotion/commit/47262b64) - Fix labels from stack traces in some cases

## 10.0.15

### Patch Changes

- [188dc0e7](https://github.com/emotion-js/emotion/commit/188dc0e7) [#1443](https://github.com/emotion-js/emotion/pull/1443) Thanks [@mitchellhamilton](https://github.com/mitchellhamilton)! - Extract styles rather than inline them in compat mode with the Global component
  - [4a3b18a2](https://github.com/emotion-js/emotion/commit/4a3b18a2) [#1451](https://github.com/emotion-js/emotion/pull/1451) Thanks [@Andarist](https://github.com/Andarist)! - Fixed issue with auto-labelling crashing on \$ at runtime

## 10.0.14

### Patch Changes

- [c0eb604d](https://github.com/emotion-js/emotion/commit/c0eb604d) [#1419](https://github.com/emotion-js/emotion/pull/1419) Thanks [@mitchellhamilton](https://github.com/mitchellhamilton)! - Update build tool
