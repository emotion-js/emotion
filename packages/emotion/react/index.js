module.exports = require('react-emotion')
