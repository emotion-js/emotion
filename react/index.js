module.exports = require('../lib/react')
