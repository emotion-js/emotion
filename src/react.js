import { Component, createElement as h } from 'react'
import { createTheming } from 'theming'
import PropTypes from 'prop-types'
import map from '@arr/map'
import { css } from './index'
import { omit } from './utils'

const theming = createTheming('__emotion__')
const { channel: CHANNEL, withTheme, ThemeProvider } = theming
export { CHANNEL, withTheme, ThemeProvider }

export default function (tag, cls, vars, content) {
  vars = vars || []

  if (!tag) {
    throw new Error(
      'You are trying to create a styled element with an undefined component.\nYou may have forgotten to import it.'
    )
  }

  class Styled extends Component {
    state = {
      theme: {}
    }

    componentWillMount () {
      if (this.context[CHANNEL]) {
        this.setState({ theme: this.context[CHANNEL].getState() })
      }
    }

    componentDidMount () {
      if (this.context[CHANNEL]) {
        this.unsubscribe = this.context[CHANNEL].subscribe(this.setTheme)
      }
    }

    componentWillUnmount () {
      if (typeof this.unsubscribe === 'function') {
        this.unsubscribe()
      }
    }

    render () {
      const { props, state } = this
      const mergedProps = {
        ...props,
        theme: state.theme
      }

      const getValue = v =>
        v && typeof v === 'function'
          ? v.cls || v(mergedProps)
          : v

      const className = css(map(cls, getValue), map(vars, getValue), content)
      console.log(className)
      mergedProps.ref = props.innerRef
      mergedProps.className = props.className
        ? className + ' ' + props.className
        : className

      return h(tag, omit(mergedProps, ['innerRef', 'theme']))
    }

    setTheme = theme => this.setState({ theme })
  }

  Styled.contextTypes = {
    [CHANNEL]: PropTypes.object
  }

  const name = typeof cls[0] === 'string' ? cls[0].split('-')[1] : ''
  const componentTag = tag.displayName || tag.name || 'Component'
  Styled.displayName = `styled(${componentTag}${name})`
  Styled.cls = '.' + cls
  Styled.spec = {
    cls,
    vars,
    content
  }
  return Styled
}
