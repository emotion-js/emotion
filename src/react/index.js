import { createElement as h } from 'react'
import { css } from '../index'
import { map, reduce, assign, omit } from '../utils'
import propsRegexString from /* preval */ './props'

export {
  flush,
  css,
  injectGlobal,
  fontFace,
  keyframes,
  hydrate,
  objStyle
} from '../index'

const push = (obj, items) => Array.prototype.push.apply(obj, items)

const reactPropsRegex = new RegExp(propsRegexString)
const testOmitPropsOnStringTag = key => reactPropsRegex.test(key)
const testOmitPropsOnComponent = key => key !== 'theme' && key !== 'innerRef'

export default function(tag, cls, objs, vars = [], content) {
  if (!tag) {
    throw new Error(
      'You are trying to create a styled element with an undefined component.\nYou may have forgotten to import it.'
    )
  }

  const componentTag = tag.displayName || tag.name || 'Component'
  const spec = {
    vars,
    content,
    objs,
    tag,
    cls
  }
  const newSpec =
    tag.__emotion_spec !== undefined ? tag.__emotion_spec.concat(spec) : [spec]
  const localTag = newSpec[0].tag

  const omitFn =
    typeof localTag === 'string'
      ? testOmitPropsOnStringTag
      : testOmitPropsOnComponent
  function Styled(props, context) {
    const getValue = v => {
      if (v && typeof v === 'function') {
        if (v.__emotion_class !== undefined) {
          return `& .${v.__emotion_class}`
        }
        return v(props, context)
      }

      return v
    }

    let finalObjs = []

    push(
      finalObjs,
      reduce(
        newSpec,
        (accum, spec) => {
          push(accum, spec.objs)
          if (spec.content) {
            push(accum, spec.content.apply(null, map(spec.vars, getValue)))
          }
          accum.push(spec.cls)
          return accum
        },
        []
      )
    )

    if (props.className) {
      push(finalObjs, props.className.split(' '))
    }

    const className = css(map(finalObjs, getValue))

    return h(
      localTag,
      omit(
        assign({}, props, {
          ref: props.innerRef,
          className
        }),
        omitFn
      )
    )
  }

  Styled.displayName = `styled(${componentTag})`
  Styled.__emotion_spec = newSpec
  Styled.__emotion_class = cls
  return Styled
}
