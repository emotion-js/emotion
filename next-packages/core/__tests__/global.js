// @flow
import 'test-utils/dev-mode'
import * as React from 'react'
import { render, unmountComponentAtNode } from 'react-dom'
import { Global } from '@emotion/core'

beforeEach(() => {
  // $FlowFixMe
  document.head.innerHTML = ''
  // $FlowFixMe
  document.body.innerHTML = `<div id="root"></div>`
})

test('basic', () => {
  render(
    <Global
      styles={{
        html: {
          backgroundColor: 'hotpink'
        },
        '@font-face': {
          fontFamily: 'some-name'
        }
      }}
    />,
    // $FlowFixMe
    document.getElementById('root')
  )
  expect(document.head).toMatchSnapshot()
  expect(document.body).toMatchSnapshot()
  unmountComponentAtNode(document.getElementById('root'))
  expect(document.head).toMatchSnapshot()
  expect(document.body).toMatchSnapshot()
})
