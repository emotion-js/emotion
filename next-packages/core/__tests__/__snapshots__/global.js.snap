// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
<head>
  <style
    data-emotion="css-global"
  >
    
    html{background-color:hotpink;}
  </style>
</head>
`;

exports[`basic 2`] = `
<body>
  <div
    id="root"
  />
</body>
`;

exports[`basic 3`] = `<head />`;

exports[`basic 4`] = `
<body>
  <div
    id="root"
  />
</body>
`;
